File: baseDatos/conectarBD.py
import pymysql
#conecta a la base de datos
class dbConnection():
	def __init__(self, dbHost, username,password,dbName):
		self.dataBaseHost = dbHost
		self.username = username
		self.password = password
		self.dataBaseName = dbName
	def connect(self):
		try:
			self.connection = pymysql.connect(self.dataBaseHost, self.username, self.password, self.dataBaseName)
		except:
			print("error al conectar a base de datos")
	def disconnect(self):
		try:
			self.connection.close()
		except:
			print("error al desconectar a la base de datos")
	def execute(self,instruction): 
		cursor = self.connection.cursor()
		result = ''
		try:
			cursor.execute(instruction)
			self.connection.commit()
			result = cursor.fetchall()
		except:
			print("error al enviar instruction")
			self.connection.rollback()
		return result

	def executeOneResult(self,instruction): 
		cursor = self.connection.cursor()
		result = ''
		try:
			cursor.execute(instruction)
			self.connection.commit()
			result = cursor.fetchone()
		except:
			print("error al enviar instruction")
			self.connection.rollback()
		return result
"""
dbHost = 'localhost'
username = 'root'
password = 'password'
dataBaseName = 'dbPT'
connection = dbConnection(dbHost, username,password,dataBaseName)
connection.connect()
a = "select * from users"
c = "otraAlejandra"
d = "algo"
#b = sql = "INSERT INTO users(idusers, nombre, password) VALUES (5,'{0}','{1}')".format(c,d)
b = "select * from users"
#print(format(connection.execute(b) )   )
resultado = connection.execute(b)
#print(resultado[1])

connection.disconnect()
"""

File: pt3/conectarBD.php
<?php

class dbConnection
{
	private $dbHost;
	private $user;
	private $pass;
	private $dbName;
	private $conn;
	public function __construct($dbHost, $user, $pass, $dbName)
	{
		$this->dbHost = $dbHost;
		$this->user = $user;
		$this->pass = $pass;
		$this->dbName = $dbName;
	}
	public function connect()
	{
		$this->conn = new mysqli($this->dbHost, $this->user, $this->pass, $this->dbName);
		if(($this->conn)->connect_error)
		{
			die("Connection failed %s\n".$this->conn->connect_error."\n");
		}
			
	}

	public function execute($query)
	{
		$result = mysqli_query($this->conn,$query);
		if(!$result)
		{
			die( "Error---- ".$query." ".($this->conn)->error."\n");
		}
		return $result;
	}
	public function disconnect()
	{
		$this->conn->close();
	}
}

//ejemplo
$db = new dbConnection("localhost","root","password","mydb");
/*
$db->connect();

$db->execute("insert into users(nombre, password) values('alejandra2','algo')");
$db->execute("insert into users(nombre, password) values('alejandra3','algo')");
$db->disconnect();
$result = $db->execute("select * from ugsers;");
while($row = $result->fetch_assoc())
{
	echo $row["nombre"]." ".$row["password"]."\n";
	//echo $row["password"];
}
$db->disconnect();
*/
?>
